opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp

```diff
@@ -351,11 +358,8 @@ class SimulatorFullyImplicitBlackoilEbos
             // update timing.
             report.solver_time += solver_timer.secsSinceStart();
 
-            if ( output_writer_.output() ) {
-                if ( output_writer_.isIORank() )
-                {
-                    stepReport.reportParam(tstep_os);
-                }
+            if ( tstep_os.is_open() ) {
```

Comment:
I would think that it is a little clearer to have the same nested if-s here, and perhaps an assert() that the file then is open? Not going to require it, just a suggestion.